dynawo/sources/Solvers/FixedTimeStep/DYNSolverCommonFixedTimeStep.cpp

```diff
@@ -426,10 +426,12 @@ bool SolverCommonFixedTimeStep::setupNewAlgRestoration(modeChangeType_t modeChan
   if (modeChangeType == ALGEBRAIC_MODE) {
     solverKINAlgRestoration_->setupNewAlgebraicRestoration(fnormtolAlg_, initialaddtolAlg_, scsteptolAlg_, mxnewtstepAlg_, msbsetAlg_, mxiterAlg_,
                                                            printflAlg_);
+    setDifferentialVariablesIndices();
```

Comment:
WHy calling it in the simplified solver? Moreover I don't see where is the declaration of this method? (I only see it in SolverIDA.h)

Reply:
The method already exist in SolverCommonFixedTimeStep

Reply:
The `self.addLine("  data->simulationInfo->discreteCall = 1;\n")` in `fill_evalDynamicYType_omc` was a bad solution the this issue. And after some more debug as long as evalDynamicYType is called after a rotateBuffer it is ok and the `discreteCall` is not needed. But it is tricky nonetheless !!!!!